src/tools/pkcs11-register.c

```diff
@@ -235,11 +235,6 @@ add_module_firefox(const char *module_path, const char *module_name, const char
 #endif
 	};
 
-	if (0 == strcmp(module_path, default_pkcs11_provider)) {
-		module_path = default_onepin_pkcs11_provider;
-		exclude_module_path = default_pkcs11_provider;
-	}
-
```

Comment:
did you try what will happen if any old `pkcs11.txt` still has

```
library=/usr/lib64/onepin-opensc-pkcs11.so
name=OpenSC smartcard framework (0.22)
```

One would expect the old entry to be removed and replaced with a new one.

I guess you did, but i really do not trust this application since it is very broken.

When i just looked at my firefox config i again found a conflict ...

```
library=/usr/lib64/p11-kit-proxy.so
name=my-nss-p11-kit-proxy.so
library=/usr/lib64/onepin-opensc-pkcs11.so
name=OpenSC smartcard framework (0.22)
```

where `my-nss-p11-kit-proxy.so` deals with the smartcard and `pkcs11-register` is doing all the wrong things

Reply:
This change is related to *always* adding `opensc-pkcs11.so` instead of `onepin-opensc-pkcs11.so` regardless of the hosting app since the application is the same with this PR. this change is not related to limitations of pkcs11-register. If you have new input regarding the latter, please use the existing issues or a new issue.